### DISCUSSION TITLE
invalid model due to "sum of all variable domains do not fit on an int64_t"

### DISCUSSION THREAD

**Discussion by gregy4** [google/or-tools#4179] [CP-SAT questions]
Hello,
  I found checking of a model for my specific geometric problem too restrictive and done too early.

  1. All variable domains are calculated including removed variables during presolve.
  2. For helper variables used only in objective function (in my case multiplication requires extra variable for a result) real domain can be also identified during presolve.

  From a user point of view it can be quite tricky to fulfill limit done by sum of all variables domains for some problems without duplication of work done by presolve.

  Are there plans to improve this functionality?

Thanks,
Jan

**Comment by lperron** [comment]
Not in the short term
